ingest.py

```diff
@@ -164,4 +244,7 @@ def main():
 
 
 if __name__ == "__main__":
+
+    source_directory, persist_directory = prompt_user()
```

Comment:
`db.persist()` doesn't want to store parquet files when I run it for the 1st time, only if a folder is previously created in the `dbs`, then it creates embeddings as it should. 
I suspect that the rest of the program is not aware that a new directory is created ?? Hm... This is working ok for you?

Reply:
Found out where's the problem. It is in `constants.py` file, where: 

```python
CHROMA_SETTINGS = Settings(
    chroma_db_impl='duckdb+parquet',
    persist_directory=PERSIST_DIRECTORY,
    anonymized_telemetry=False
)
```
Is always holding the previous `PERSIST_DIRECTORY` from the env variable. And, if you change the directory of ingest it will write chroma files to the old folder from `.env` variable. So the idea is to remove this constant from here and make it dynamic on change because `set_key('.env', 'YOUR_ENV_VAR', path)` will only store / change value in .env file when ingestion is done

Reply:
It's an interesting bug, it doesn't seem to arise for me starting with a clean clone of the repo, but I can see that it could arise if things change without calling ingest.py before inference.

I'm not familiar with updating class parameters on environmental changes, I tried a few methods and can't seem to get the CHROMA_SETTINGS parameter to update. Maybe you have an approach?

Here is what I tried doing in the `constants.py` file:

```python
import os
from dotenv import load_dotenv
from chromadb.config import Settings

class DynamicSettings(Settings):
    @property
    def persist_directory(self):
        return self.get_persist_directory()

    def get_persist_directory():
        load_dotenv(override=True)
        return os.environ.get('PERSIST_DIRECTORY')

# Define the Chroma settings
CHROMA_SETTINGS = DynamicSettings(
    chroma_db_impl='duckdb+parquet',
    anonymized_telemetry=False,
)
```

Reply:
Thanks for the tips, I ended up changing CHROMA_SETTINGS from a constant to a function which would dynamically update the persist directory when called, and CHROMA_SETTINGS occurences in the code. From my testing this seems to solve the error that you've been getting.

Reply:
Yea, I was confused why is this happening to me until I changed it at both places, you know that other place is here:

```python
Chroma(persist_directory=persist_directory........
```

so maybe if we commit to setting persistence inside CHROMA_SETTINGS, might help ... but glad it work now